scipy/spatial/transform/rotation.py

```diff
@@ -1460,3 +1461,109 @@ def __getitem__(self, indexer):
+            vectors. If None, then all values in `weights` are assumed to be
+            equal. Default is None.
+        normalized : boolean, optional
+            If True, assume input vectors `a` and `b` to have unit norm. If
+            False, normalize `a` and `b` before estimating rotation. Default
+            is False.
+
+        Returns
+        -------
+        estimated_rotation : `Rotation` instance
+            Best estimate of the rotation that transforms `b` to `a`.
+        covariance_matrix : `numpy.ndarray`, shape (3, 3)
+            Covariance matrix of the three component error vector of the Euler
+            angles describing the `estimated_rotation`.
+
+        References
+        ----------
+        .. [1] F. Landis Markley,
+                “Attitude determination using vector observations: a fast
+                optimal matrix algorithm,” Journal of Astronautical Sciences,
+                Vol. 41, No.2, 1993, pp. 261-280.
+        """
+        outvecs = np.asarray(a)
+        if outvecs.ndim != 2 or outvecs.shape[-1] != 3:
+            raise ValueError("Expected input `a` to have shape (N, 3), "
+                             "got {}".format(outvecs.shape))
+        vectors = np.asarray(b)
+        if vectors.ndim != 2 or vectors.shape[-1] != 3:
+            raise ValueError("Expected input `b` to have shape (N, 3), "
+                             "got {}.".format(vectors.shape))
+
+        if outvecs.shape != vectors.shape:
+            raise ValueError("Expected inputs `a` and `b` to have same shapes"
+                             ", got {} and {} respectively.".format(
+                                outvecs.shape, vectors.shape))
+
+        if vectors.shape[0] == 1:
+            raise ValueError("Rotation cannot be estimated using a single "
+                             "vector.")
+
+        if weights is None:
+            weights = np.ones(vectors.shape[0])
+        else:
+            weights = np.asarray(weights)
+            if weights.ndim != 1:
+                raise ValueError("Expected `weights` to be 1 dimensional, got "
+                                 "shape {}.".format(weights.shape))
+            if weights.shape[0] != vectors.shape[0]:
+                raise ValueError("Expected `weights` to have number of values "
+                                 "equal to number of input vectors, got "
+                                 "{} values and {} vectors.".format(
+                                    weights.shape[0], vectors.shape[0]))
+        weights = weights / scipy.linalg.norm(weights)
+
+        if not normalized:
+            outvecs = outvecs / scipy.linalg.norm(outvecs, axis=1)[:, None]
+            vectors = vectors / scipy.linalg.norm(vectors, axis=1)[:, None]
+
+        B = np.einsum('ji,jk->ik', weights[:, None] * outvecs, vectors)
+        u, s, vh = np.linalg.svd(B)
+
+        zeta = (s[0]+s[1]) * (s[1]+s[2]) * (s[2]+s[0])
+        # TODO: Maybe change to close to zero?
```

Comment:
Hard to say, you can experiment with two vectors being very close to parallel. Find out when estimation becomes unstable and what are the values of zeta in this scenario.